src/Workspaces/Core/Portable/Rename/Renamer.SyncNamespaceDocumentAction.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ChangeNamespace;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Rename
+{
+    public static partial class Renamer
+    {
+        /// <summary>
+        /// Action that will sync the namespace of the document to match the folders property 
+        /// of that document
+        /// </summary>
+        internal class SyncNamespaceDocumentAction : RenameDocumentAction
+        {
+            private readonly IReadOnlyList<string> _newFolders;
+            private readonly AnalysisResult _analysis;
```

Comment:
Want to make this inner class naming be consistent with the other one? (The other one was RenameSymbolDocumentAnalysis" which was a bit redundant with the parent class and said the same thing twice.)